src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1533,6 +1534,22 @@ InferenceProfiler::DetermineStatsModelVersion(
   return cb::Error::Success;
 }
 
+// Only for unit-testing
+#ifndef DOCTEST_CONFIG_DISABLE
+void
+InferenceProfiler::SetTopLevelRequestCaching(
+    bool enable_top_level_request_caching)
+{
+  // parser_ = std::make_shared<MockModelParser>();
+  parser_ = std::make_shared<ModelParser>(cb::BackendKind::TRITON);
+  if (parser_ == nullptr) {
+    std::cout << "NULL Pointer" << std::endl;
```

Comment:
should this be a print? or some kind of assertion or exception to fail the test?

Reply:
I added this as a print for me to figure out if  parser_ can be null due to some error.
Now that I think about it, I should add an assertion in my unit test to fail when parser_ is null_ptr.